data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java

```diff
@@ -29,6 +30,7 @@
     private X509Certificate x509Certificate;
     private PrivateKey privateKey;
     private AuthenticationType authenticationType;
+    private AuthenticationResult lastAuthenticationResult;
```

Comment:
Can you please hold an instance of AuthorizationContext as a class member, and not the result?
We do the same in .Net, python, and node.
See an example here:
https://github.com/Azure/azure-kusto-python/blob/master/azure-kusto-data/azure/kusto/data/security.py